internal/statemachine/classic_states.go

```diff
@@ -383,6 +353,50 @@ func (stateMachine *StateMachine) installPackages() error {
 	return nil
 }
 
+func (stateMachine *StateMachine) gatherPackages(imageDef *imagedefinition.ImageDefinition) {
+	if imageDef.Customization != nil {
+		for _, packageInfo := range imageDef.Customization.ExtraPackages {
+			stateMachine.Packages = append(stateMachine.Packages,
+				packageInfo.PackageName)
+		}
+	}
+
+	// Make sure to install the extra kernel if it is specified
+	if imageDef.Kernel != "" {
+		stateMachine.Packages = append(stateMachine.Packages,
+			imageDef.Kernel)
+	}
+}
+
+// generateMountPointCmds generate lists of mount/umount commands for a list of mountpoints
+func generateMountPointCmds(mountPoints []*mountPoint, scratchDir string) (allMountCmds []*exec.Cmd, allUmountCmds []*exec.Cmd, err error) {
+	for _, mp := range mountPoints {
+		var mountCmds, umountCmds []*exec.Cmd
+		var err error
+		if mp.bind {
+			mp.src, err = osMkdirTemp(scratchDir, strings.Trim(mp.relpath, "/"))
+			if err != nil {
+				return nil, nil, fmt.Errorf("Error making temporary directory for mountpoint \"%s\": \"%s\"",
+					mp.relpath,
+					err.Error(),
+				)
+			}
+		}
+
+		mountCmds, umountCmds, err = mp.getMountCmd()
+		if err != nil {
+			return nil, nil, fmt.Errorf("Error preparing mountpoint \"%s\": \"%s\"",
+				mp.relpath,
+				err.Error(),
+			)
+		}
+
+		allMountCmds = append(allMountCmds, mountCmds...)
+		allUmountCmds = append(umountCmds, allUmountCmds...)
+	}
+	return
```

Comment:
The things you want to return are missing here (just looked at the code and you fixed it somewhere in the next commits)

Reply:
Yes, this is one of the rare cases where I fixed this function in another commit. In go you can have "naked returns" by giving names of returned values in the function signature and only call return. This is handy but considered confusing when the function is too long. Currently the threshold for the linter I use is 5 lines (for the whole function). So this code was working fine, but I added back the returned values names.